src/Symfony/Component/EventDispatcher/InvokableEventSubscriberTrait.php

```diff
@@ -0,0 +1,56 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Component\EventDispatcher;
+
+use Symfony\Contracts\EventDispatcher\Event as ContractsEvent;
+
+/**
+ * Helper trait for single-event subscription.
+ *
+ * @author Yonel Ceruto <yonelceruto@gmail.com>
+ */
+trait InvokableEventSubscriberTrait
+{
+    /**
+     * {@inheritdoc}
+     */
+    final public static function getSubscribedEvents(): array
+    {
+        return [static::getEventName() => ['__invoke', static::getPriority()]];
```

Comment:
This should use `self::` instead of `static::`: now that it's a trait, this is all internal details - not extensibility hook. That's also why new methods should be private: we don't want to impose constraints to child classes. The trait is pure implementation details, that's it's beauty.

Reply:
I don't understand why the extensibility hook is a problem. If developer want to change the priority of the event, they can (thank to the IDE) discover the right method just by pressing Ctrl+space.

Reply:
I'd prefer be pragmatic here, extensibility hook cannot be avoided with traits, so let's improve the DX.

Reply:
sorry but that's wrong from a design pov, see my comment above about phpstorm, which is buggy here